radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/AtomStore.java

```diff
@@ -48,4 +43,23 @@
 	 * @return a never ending observable of updates
 	 */
 	Observable<AtomObservation> getAtomObservations(RadixAddress address);
+
+	/**
+	 * Retrieve the current set of validated up particles at a given shardable.
+	 * @param address the address to get the particles under
+	 * @return a stream of all up particles of the current local view
+	 */
+	Stream<Particle> getUpParticles(RadixAddress address, String uuid);
+
+	/**
+	 * Adds the particle group to the staging area
+	 * @param particleGroup the particle group to add to staging area
+	 */
+	void stageParticleGroup(String uuid, ParticleGroup particleGroup);
+
+	/**
+	 * Retrieves all staged particle groups and clears
+	 * @return all staged particle groups in order
```

Comment:
What order?

Reply:
order of insertion I would presume, but please clarify